config/manager/manager.yaml

```diff
@@ -26,19 +26,39 @@ spec:
     spec:
       affinity:
         nodeAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-            nodeSelectorTerms:
-              - matchExpressions:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 3
+              preference:
+                matchExpressions:
+                  - key: node-role.kubernetes.io/infra
+                    operator: In
+                    values: 
+                    - reserved
+            - weight: 1
+              preference:
+                matchExpressions:
                   - key: node-role.kubernetes.io/master
-                    operator: Exists
-              - matchExpressions:
+                    operator: In
+            - weight: 1
+              preference:
+                matchExpressions:
                   - key: node-role.kubernetes.io/control-plane
-                    operator: Exists
+                    operator: In
       tolerations:
         - effect: NoSchedule
           key: node-role.kubernetes.io/master
+          operator: Exists
         - effect: NoSchedule
           key: node-role.kubernetes.io/control-plane
+          operator: Exists
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/infra
+          operator: Exists
+          value: reserved
```

Comment:
`value` does not make sense with `Exists`, only with `Equal`. But using no value is better as suggested above, I'd remove `value`